src/envoy/http/backend_auth/filter.cc

```diff
@@ -43,9 +43,9 @@ typedef ConstSingleton<RcDetailsValues> RcDetails;
 
 }  // namespace
 
-FilterHeadersStatus Filter::decodeHeaders(HeaderMap& headers, bool) {
+FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool) {
   absl::string_view operation = Utils::getStringFilterState(
-      decoder_callbacks_->streamInfo().filterState(), Utils::kOperation);
+      *decoder_callbacks_->streamInfo().filterState(), Utils::kOperation);
```

Comment:
Should we change `Utils::getStringFilterState` to take in a shared pointer instead of dereferencing everytime?

Reply:
In general, if a function is not going to take ownership of object, it is better not to pass in share_ptr.